src/index.ts

```diff
@@ -2,7 +2,8 @@ import axios from 'axios'
 import type { AxiosInstance, AxiosRequestConfig } from 'axios'
 import jwtDecode from 'jwt-decode'
 import type { JwtPayload } from 'jwt-decode'
-import Storage from './storage'
+import { StorageProxy, StorageServiceType } from './StorageProxy'
+import './storage'
```

Comment:
I don't recommend using side effects here to initialize - I suggest exporting a function and calling it to init storage. Less ambiguity with tree-shaking and it won't break if you import this file in a non-web context since it accesses localStorage

Reply:
I thought about it also. But if I do it (remove the initialization which is in ./storage currently). Then it will be a breaking change in the library. And in the upgdate path every users would need to initialize storage by themselves.

I did not do anything extra what was not in the library already. If you give a green light I could ditch ./storage import thus making manual initialization mandatory

Read: non-web context was already indeed broken due to access to window.localStorage :)